packages/react/src/ReactStartTransition.js

```diff
@@ -48,12 +43,15 @@ export function startTransition(
   if (enableAsyncActions) {
     try {
       const returnValue = scope();
+      const onStartTransitionFinish = ReactSharedInternals.S;
+      if (onStartTransitionFinish !== null) {
+        onStartTransitionFinish(transition, returnValue);
+      }
```

Comment:
Do we need to account for cases like

```
require('react').startTransition(() => {
  require('react-dom').createRoot(...).render(<App />)
})
```

I suppose the shared internals would be mutated by the time we reference the onStartTransitionFinish but maybe if you did the require after something async it would be too late?

Reply:
Unlike with flushSync I don't think there's a great way to account for the async case. The non-async case also wouldn't work if we were to move the `isTransition` logic off the shared internals and into the reconciler only, which is probably what we'll eventually do. I think it's fine to leave as a wont-fix.